cmd/noms/diff/diff_test.go

```diff
@@ -237,85 +342,131 @@ func TestNomsListDiff(t *testing.T) {
 +   "a1": "a-one-diff"
   }
 `
+	expectedPaths3 := []string{
+		`[2]["m3"]`,
+		`[2]["m4"]["a1"]`,
+	}
 
 	l5 := createList(mm1, mm2, mm3, mm4)
 	l6 := createList(mm1, mm2, mm3x, mm4)
 
 	tf := func(leftRight bool) {
 		buf := &bytes.Buffer{}
-		Diff(buf, l1, l2, leftRight)
+		PrintDiff(buf, l1, l2, leftRight)
 		assert.Equal(expected1, buf.String())
 
+		paths := pathsFromDiff(l1, l2, leftRight)
+		assert.Equal(expectedPaths1, paths)
+
 		buf = &bytes.Buffer{}
-		Diff(buf, l3, l4, leftRight)
+		PrintDiff(buf, l3, l4, leftRight)
 		assert.Equal(expected2, buf.String())
 
-		buf = &bytes.Buffer{}
-		Diff(buf, l5, l6, leftRight)
+		paths = pathsFromDiff(l3, l4, leftRight)
+		assert.Equal(expectedPaths2, paths)
 
+		buf = &bytes.Buffer{}
+		PrintDiff(buf, l5, l6, leftRight)
 		assert.Equal(expected3, buf.String())
+
+		paths = pathsFromDiff(l5, l6, leftRight)
+		assert.Equal(expectedPaths3, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
 
-func TestNomsBlobDiff(t *testing.T) {
+func TestNomsDiffPrintBlob(t *testing.T) {
 	assert := assert.New(t)
 
 	expected := "-   Blob (2.0 kB)\n+   Blob (11 B)\n"
+	expectedPaths1 := []string{``}
 	b1 := types.NewBlob(strings.NewReader(strings.Repeat("x", 2*1024)))
 	b2 := types.NewBlob(strings.NewReader("Hello World"))
 
 	tf := func(leftRight bool) {
 		buf := &bytes.Buffer{}
-		Diff(buf, b1, b2, leftRight)
+		PrintDiff(buf, b1, b2, leftRight)
 		assert.Equal(expected, buf.String())
+
+		paths := pathsFromDiff(b1, b2, leftRight)
+		assert.Equal(expectedPaths1, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
 
-func TestNomsTypeDiff(t *testing.T) {
+func TestNomsDiffPrintType(t *testing.T) {
 	assert := assert.New(t)
 
 	expected1 := "-   List<Number>\n+   List<String>\n"
+	expectedPaths1 := []string{``}
```

Comment:
Using a raw string instead of a normal string (i.e. `''`) is a bit weird but ok.